desktop/src/com/mygdx/game/client/network/ComponentMessageListener.java

```diff
@@ -20,39 +24,61 @@
 @Log
 public class ComponentMessageListener extends AbstractWebSocketListener {
 
-  /**
-   * Used as default value when invoking {@link ObjectMap#get(Object, Object)} on {@link #handlers} to prevent
-   * NPE.
-   */
-  private static final Handler<Component> unknown = (webSocket, packet) -> NOT_HANDLED;
-  private final ObjectMap<Class<?>, Handler<Component>> handlers = new ObjectMap<>();
-  private final Json json = new Json();
-  private final ComponentMapper<Position> positionMapper;
+  private static final Handler NO_HANDLER = (webSocket, worldEntity, packet) -> {
+    throw new RuntimeException("Every ComponentMessage needs to be handled: id=" + worldEntity
+        + " content=" + packet);
+  };
+  private final ObjectMap<Class<? extends Component>, Handler> handlers = new ObjectMap<>();
+  private final NetworkWorldEntityMapper networkWorldEntityMapper;
 
   @Inject
   public ComponentMessageListener(
-      NetworkEntityManager networkEntityManager,
-      World world
+      NetworkWorldEntityMapper networkWorldEntityMapper,
+      World world,
+      GameScreenAssets assets,
+      FieldFactory fieldFactory,
+      UnitFactory unitFactory,
+      GameState gameState
   ) {
-    this.positionMapper = world.getMapper(Position.class);
+    this.networkWorldEntityMapper = networkWorldEntityMapper;
+    final var positionMapper = world.getMapper(Position.class);
+    final var coordinatesMapper = world.getMapper(Coordinates.class);
 
-    registerHandler(Name.class, (webSocket, packet) -> {
-      var entity = networkEntityManager.getWorldEntity(packet.getEntityId());
-      if (entity == -1) {
-        throw new RuntimeException("entity needs to be created locally before use");
+    registerHandler(EntityConfigId.class, (webSocket, worldEntity, packet) -> { // todo zmienić znowu na serwisy i fabryki,
+      // todo bo to jednak nie jest komponent tylko pojedyncza wiadomość xD
+      // albo coś
+      var entityConfigId = ((EntityConfigId) packet).getId();
+      if (entityConfigId == 1) { // 1, 2 są z plików jsona EntityConfigów
+        log.info("field id " + worldEntity);
```

Comment:
No narazie może tak być, ale potem zmienię to w moim tickecie w którym dodałem więcej unitów